File: designtime/webstudio/ui/src/app/editors/rule-template-instance-builder/RangeFilterValueImpl.ts
/**
 * @Author: Rahil Khera
 * @Date:   2017-04-20T10:04:35+05:30
 * @Last modified by:   Rahil Khera
 * @Last modified time: 2017-04-20T10:04:38+05:30
 */

import { FilterValue } from './FilterValue';

export class RangeFilterValueImpl extends FilterValue {

  private minValue: string = undefined;
  private maxValue: string = undefined;
  getMinValue(): string {
    return this.minValue;
  }
  setMinValue(value: string) {
    this.minValue = value;
  }

  getMaxValue(): string {
    return this.maxValue;
  }
  setMaxValue(value: string) {
    this.maxValue = value;
  }
}

File: common/com.tibco.cep.studio.common/src/com/tibco/cep/webstudio/model/rule/instance/impl/RangeFilterValueImpl.java
package com.tibco.cep.webstudio.model.rule.instance.impl;

import com.tibco.cep.webstudio.model.rule.instance.IInstanceChangedEvent;
import com.tibco.cep.webstudio.model.rule.instance.InstanceChangedEvent;
import com.tibco.cep.webstudio.model.rule.instance.RangeFilterValue;

public class RangeFilterValueImpl extends RuleTemplateInstanceObject implements RangeFilterValue {

	/**
	 * 
	 */
	private static final long serialVersionUID = -2510139816666664557L;
	/**
	 * 
	 */
	private String minValue;
	private String maxValue;

	@Override
	public String getMinValue() {
		return this.minValue;
	}

	@Override
	public String getMaxValue() {
		return this.maxValue;
	}
	
	@Override
	public void setMinValue(String value) {
		this.minValue = value;
		InstanceChangedEvent changeEvent = new InstanceChangedEvent(IInstanceChangedEvent.CHANGED, MIN_FEATURE_ID, this, value);
		fireInstanceChanged(changeEvent);
	}

	@Override
	public void setMaxValue(String value) {
		this.maxValue = value;
		InstanceChangedEvent changeEvent = new InstanceChangedEvent(IInstanceChangedEvent.CHANGED, MAX_FEATURE_ID, this, value);
		fireInstanceChanged(changeEvent);
	}
}
